src/Domain/Creators/CylindricalBinaryCompactObject.hpp

```diff
@@ -434,10 +354,7 @@ class CylindricalBinaryCompactObject : public DomainCreator<3> {
       const Options::Context& context = {});
 
   CylindricalBinaryCompactObject(
-      double initial_time, ExpansionMapOptions expansion_map_options,
-      std::array<double, 3> initial_angular_velocity,
-      std::array<double, 3> initial_size_map_values_A,
-      std::array<double, 3> initial_size_map_values_B,
+      bco::TimeDependentMapOptions time_dependent_options,
```

Comment:
So I tried to do this originally (because I agree it'd be a lot simpler) but TL;DR it won't work because of options.

----

The reason it won't work is because we have two different sets of "optional" options, meaning boundary conditions and time dependent maps. These are compile-time options to the `using options =` list. So you have 4 different combos of things that all technically need their own constructor:

1. With BC, with TimeDep
2. With BC, without TimeDep
3. Without BC, with TimeDep
4. Without BC, without TimeDep

So if I have a general constructor like
```cpp
CylBCO(type common_opts,
       TimeDepOpts time_dep_opts = std::nullopt,
       BCs boundary_condtions = nullptr)
```

this constructor works fine for 1, 3, and 4, but it doesn't work for 2 because 2 expects a constructor like
```cpp
CylBCO(type common_opts,
       BCs boundary_condtions = nullptr,
       TimeDepOpts time_dep_opts = std::nullopt)
```

so you basically have to have a second constructor at that point.

Reply:
Ok that's annoying but don't worry about it here then